onboard-server/src/main/java/no/vegvesen/ixn/serviceprovider/OnboardRestController.java

```diff
@@ -300,59 +302,106 @@ public GetSubscriptionResponse getServiceProviderSubscription(@PathVariable Stri
 	}
 
 	@RequestMapping(method = RequestMethod.POST, path = "/{serviceProviderName}/privatechannels", produces = MediaType.APPLICATION_JSON_VALUE)
-	public PrivateChannelApi addPrivateChannel(@PathVariable String serviceProviderName, @RequestBody PrivateChannelApi clientChannel) {
+	public AddPrivateChannelResponse addPrivateChannel(@PathVariable String serviceProviderName, @RequestBody AddPrivateChannelRequest clientChannel) {
 		OnboardMDCUtil.setLogVariables(nodeProperties.getName(), serviceProviderName);
 		logger.info("Add private channel for service provider {}", serviceProviderName);
 		this.certService.checkIfCommonNameMatchesNameInApiObject(serviceProviderName);
 
-		if(clientChannel == null) {
+		if (clientChannel == null || clientChannel.getPrivateChannels() == null || clientChannel.getPrivateChannels().isEmpty()) {
 			throw new PrivateChannelException("Client channel cannot be null");
```

Comment:
The message is a bit strange, the client might not know what "client channel" is, as it is an internal variable name. Also, this will return a 500 error to the client. It should return 400 bad request (it needs to be added to the error advise).